packages/astro/src/@types/astro.ts

```diff
@@ -869,28 +869,28 @@ export type Params = Record<string, string | number | undefined>;
 
 export type Props = Record<string, unknown>;
 
-type Body = string;
-
 export interface AstroAdapter {
 	name: string;
 	serverEntrypoint?: string;
 	exports?: string[];
 	args?: any;
 }
 
+type Body = string;
+
 export interface APIContext {
 	params: Params;
 	request: Request;
 }
 
-export interface EndpointOutput<Output extends Body = Body> {
-	body: Output;
+export interface EndpointOutput {
+	body: Body;
 }
 
-export type APIRoute = (context: APIContext) => EndpointOutput | Response;
+export type APIRoute = (context: APIContext) => EndpointOutput | Response | Promise<EndpointOutput | Response>;
 
 export interface EndpointHandler {
-	[method: string]: APIRoute | ((params: Params, request: Request) => EndpointOutput | Response);
```

Comment:
Any reason for changing this? This is needed for supporting the deprecated 2 param API.

Reply:
But aren't both of the types the same?

Reply:
No, that's why the build is failing. The second type takes two parameters. The APIRoute signature just takes the one.

Reply:
Oops! Sorry! Didn't even realize!